From: Duncan Eastoe <duncan@graphiant.com>
Date: Tue, 2 Aug 2022 13:56:38 +0100
Subject: common: add internal sr_val_ly2sr_with_opts() API

This will allow the addition of optional behaviours when converting
libyang data into a sysrepo value.
---
 src/common.c |  6 ++++++
 src/common.h | 18 ++++++++++++++++++
 2 files changed, 24 insertions(+)

diff --git a/src/common.c b/src/common.c
index 1a08194..e3a6501 100644
--- a/src/common.c
+++ b/src/common.c
@@ -3950,6 +3950,12 @@ sr_ev2api(sr_sub_event_t ev)
 
 sr_error_info_t *
 sr_val_ly2sr(const struct lyd_node *node, int with_origin, sr_val_t *sr_val)
+{
+    return sr_val_ly2sr_with_opts(node, SR_VAL_LY2SR_DEFAULT, sr_val);
+}
+
+sr_error_info_t *
+sr_val_ly2sr_with_opts(const struct lyd_node *node, sr_val_ly2sr_opts_t opts, sr_val_t *sr_val)
 {
     sr_error_info_t *err_info = NULL;
     char *ptr;
diff --git a/src/common.h b/src/common.h
index f6bc912..684a958 100644
--- a/src/common.h
+++ b/src/common.h
@@ -229,6 +229,24 @@ typedef struct {
     LYS_INFORMAT format;
 } sr_int_update_mod_t;
 
+/**
+ * Graphnos specific changes
+ */
+
+typedef enum sr_val_ly2sr_opts_e {
+    SR_VAL_LY2SR_DEFAULT = 0,             /**< Standard behaviour */
+} sr_val_ly2sr_opts_t;
+
+/**
+ * @brief Transform a libyang node into sysrepo value.
+ *
+ * @param[in] node libyang node to transform.
+ * @param[in] opts Options to adjust default behaviour.
+ * @param[out] sr_val sysrepo value.
+ * @return err_info, NULL on success.
+ */
+sr_error_info_t *sr_val_ly2sr_with_opts(const struct lyd_node *node, sr_val_ly2sr_opts_t opts, sr_val_t *sr_val);
+
 /*
  * From sysrepo.c
  */
